### DISCUSSION TITLE
Automatically create multiple blocks on paste content

### DISCUSSION THREAD

**Discussion by vilcudragos** [area17/twill#1221] [Q&A]
Hi guys, 
We use twill for blogs management and is great. We are looking now to increase the speed of the creation posts.
So we use the block editor and have different blocks, like headings, simple editor, images ... 

What we are looking for is a way to automate the copy-pasted content and automatically create blocks for each paragraph, list, heading ... only text content. 

So my idea is to have one row block where you copy the content like quill editor and have a validate button or something that can create automatically the blocks. Maybe what I need is to understand how can I programmatically create blocks.

This will reduce significantly the time for creating an article and also give the flexibility to add pictures in the content as all the content will be split into chunks. 

Many thanks

**Comment by pboivin** [comment]
Hi @vilcudragos,

I recently explored a similar problem where I needed to create empty blocks, here's how I did it:
```php
// in some method of your model

    app(\A17\Twill\Repositories\BlockRepository)->create([
        'blockable_id' => $this->id,
        'blockable_type' => static::class,
        'position' => 1,
        'content' => '{}',
        'type' => 'banner',
    ]);
```

This will attach an empty `banner` block to the current item.

`content` is a simple json encoded array with all your block's fields, something like:
```php
        'content' => json_encode([
            'title' => [
                'en' => 'Hello',
                'fr' => 'Bonjour',
            ],

            // ...
        ]),
```

You can probably find more relevant examples in your `blocks` database table :)

Hope this can help!
